datafusion/src/scalar.rs

```diff
@@ -1171,6 +1277,20 @@ impl FromStr for ScalarValue {
     }
 }
 
+impl From<Vec<(&str, ScalarValue)>> for ScalarValue {
+    fn from(value: Vec<(&str, ScalarValue)>) -> Self {
+        let mut fields: Vec<Field> = Vec::new();
+        let mut scalars: Vec<ScalarValue> = Vec::new();
+
+        value.iter().for_each(|(name, scalar)| {
+            fields.push(Field::new(name, scalar.get_datatype(), false));
+            scalars.push(scalar.clone());
+        });
```

Comment:
You might be able to write this in a more functional style (and avoid `mut`) using `unzip`: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.unzip

Something like (untested)

```
let (fields, scalars) = value.iter()
      .for_each(|(name, scalar)| {
            (Field::new(name, scalar.get_datatype(), false),
            scalar.clone())
        })
      .unzip();
```